File: Submissions/ManthanCodefest_017/A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {

  int n;
  cin >> n;

  map<string, int> hashmap;
  for (int i = 0; i < n; i++) {
    string v;
    cin >> v;
    if (hashmap.count(v) == 0) {
      hashmap[v] = 1;
      cout << "NO" << endl;
    } else {
      cout << "YES" << endl;
    }
  }

}

File: Submissions/ManthanCodefest_017/A/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();

    TreeMap<String, Integer> hashmap = new TreeMap<>();

    for (int i = 0; i < n; i++) {
      String v = sc.next();
      if (hashmap.containsKey(v) == false) {
        hashmap.put(v, 1);
        System.out.println("NO");
      } else {
        System.out.println("YES");
      }
    }

  }

}